src/threaded_rodeo.rs

```diff
@@ -979,8 +979,9 @@ where
             .expect("failed to allocate memory for interner");
 
         for (string, key) in deser_map {
-            if key.into_usize() > highest {
-                highest = key.into_usize();
+            let next_key = key.into_usize() + 1;
+            if next_key > highest {
+                highest = next_key;
```

Comment:
Couldn't this be replaced with incrementing `highest` after the loop?

Reply:
I think that would require having `highest` as an `Option` starting with `None`, because otherwise when deserializing an empty `Rodeo` you'd skip a key